src/rp2_common/hardware_pio/include/hardware/pio.h

```diff
@@ -1717,20 +1723,36 @@ static inline void pio_sm_clear_fifos(PIO pio, uint sm) {
  * before restoring the state machine's pin configuration to what it was.
  *
  * This method is provided as a convenience to set initial pin states, and should not be used against a state machine that is enabled.
+ * Note: This method only works for pins < 32. To use with pins >= 32 call pio_sm_set_pins64
  *
  * \param pio The PIO instance; e.g. \ref pio0 or \ref pio1
  * \param sm State machine index (0..3) to use
  * \param pin_values the pin values to set
  */
 void pio_sm_set_pins(PIO pio, uint sm, uint32_t pin_values);
 
+/*! \brief Use a state machine to set a value on all pins for the PIO instance
+ *  \ingroup hardware_pio
+ *
+ * This method repeatedly reconfigures the target state machine's pin configuration and executes 'set' instructions to set values on all 32 pins,
+ * before restoring the state machine's pin configuration to what it was.
+ *
+ * This method is provided as a convenience to set initial pin states, and should not be used against a state machine that is enabled.
+ *
+ * \param pio The PIO instance; e.g. \ref pio0 or \ref pio1
```

Comment:
... ` or \ref pio2` on an RP2350? :wink:

Reply:
The whole file is missing this. Can we please address in a separate PR if it's important?

Reply:
Sorry, I hadn't noticed that.